### PR TITLE
utils: add lock timestamps for DEBUG_THREADS

### PR DESCRIPTION
**Pull request by mbradeen**
Adds last locked and unlocked timestamps as well as a
counter for the number of times the lock has been
attempted (vs locked/unlocked) to debug output printed
using the DEBUG_THREADS option.

Resolves: #110

### CODE REVIEW

File: main/utils.c [hunk 1]
```diff
@@ -1322,16 +1331,29 @@ static void append_lock_information(struct ast_str **str, struct thr_lock_info *
...
 				   lock_info->locks[i].pending > 0 ? "Waiting for " :
 				   lock_info->locks[i].pending < 0 ? "Tried and failed to get " : "", i,
 				   lock_info->locks[i].file,
 				   locktype2str(lock_info->locks[i].type),
 				   lock_info->locks[i].line_num,
-				   lock_info->locks[i].func, lock_info->locks[i].lock_name,
+				   lock_info->locks[i].func,
+				   lock_info->locks[i].lock_name,
 				   lock_info->locks[i].lock_addr,
+				   lock_info->locks[i].last_locked.tv_sec,
```
Comment (jcolp) [inline main/utils.c:1349]: This output is already a bit unfriendly at times, for the time based stuff can it be made more user friendly in output?

Comment (mbradeen) [reply main/utils.c:1349]: maybe cut it to deltas?

**Comment by mbradeen** [comment]
cherry-pick-to: 20
cherry-pick-to: 18
cherry-pick-to: certified/18.9

File: main/utils.c (hunk 1 above)
Comment (jcolp) [reply main/utils.c:1349]: Whatever we actually think is useful, of which I'm probably not a good person to say because I rarely use this.

**Comment by jcolp** [comment]
Can you give an example of what this looks like now?

**Comment by mbradeen** [comment]
> Can you give an example of what this looks like now?

=======================================================================
=== GIT-master-4099f965e1M
=== Currently Held Locks at Time: 468334:09:29.088484 =================
=======================================================================
===
=== <pending> <lock#> (<file>): <lock type> <line num> <function> <lock name> <lock addr>
=== <locked at>, <failed at>, <held for> (attempts, times locked)
===
=== Thread ID: 0x7f8bff3ca700 LWP:30685 (lock_thread          started at [   83] test_named_lock.c named_lock_test())
=== ---> Lock #0 (test_named_lock.c): MUTEX 49 lock_thread lock 0x7f8c00004ef0
===      468334:09:29.088294, 00:00:00.000000, 00:00:00.000251 (1, 1)
	asterisk __ast_pthread_mutex_lock()
	asterisk __ao2_lock()
	[0x7f8c8dc3a449] test_named_lock.so test_named_lock.c:50 lock_thread()
	asterisk <unknown>()
	[0x7f8cd707b609] libpthread.so.0 pthread_create.c:478 start_thread()
	[0x7f8cd6cb4133] libc.so.6 :0 clone()
=== -------------------------------------------------------------------
===
=== Thread ID: 0x7f8bff34d700 LWP:30686 (lock_thread          started at [   84] test_named_lock.c named_lock_test())
=== ---> Lock #0 (test_named_lock.c): MUTEX 49 lock_thread lock 0x7f8c0c005070
===      468334:09:29.089931, 00:00:00.000000, 00:00:00.026615 (1, 1)
	asterisk __ast_pthread_mutex_lock()
	asterisk __ao2_lock()
	[0x7f8c8dc3a449] test_named_lock.so test_named_lock.c:50 lock_thread()
	asterisk <unknown>()
	[0x7f8cd707b609] libpthread.so.0 pthread_create.c:478 start_thread()
	[0x7f8cd6cb4133] libc.so.6 :0 clone()
=== -------------------------------------------------------------------
===
=======================================================================

**Comment by jcolp** [comment]
I think that didn't paste right. Using the <> code option may allow it to display.

**Review by jcolp — APPROVED** [review]

**Review by gtjoseph — APPROVED** [review]

**Pull request merged**